common/elasticsearch/client/os2/client.go

```diff
@@ -206,123 +210,101 @@ func (c *OS2) IsNotFoundError(err error) bool {
-		Body:  strings.NewReader(body),
+	req := osapi.MappingPutReq{
+		Indices: []string{index},
+		Body:    strings.NewReader(body),
 	}
 
-	resp, err := req.Do(ctx, c.client)
+	_, err := c.client.Indices.Mapping.Put(ctx, req)
 	if err != nil {
 		return fmt.Errorf("OpenSearch PutMapping: %w", err)
 	}
 
-	defer closeBody(resp)
-
-	if resp.IsError() {
-		return c.parseError(resp)
-	}
-
 	return nil
 }
 
 func (c *OS2) CreateIndex(ctx context.Context, index string) error {
-	req := osapi.IndicesCreateRequest{
+	req := osapi.IndicesCreateReq{
 		Index: index,
 	}
 
-	resp, err := req.Do(ctx, c.client)
+	_, err := c.client.Indices.Create(ctx, req)
 
 	if err != nil {
-		return fmt.Errorf("OpenSearch CreateIndex: %w", err)
-	}
-
-	defer closeBody(resp)
-
-	if resp.IsError() {
-		return c.parseError(resp)
+		return err
 	}
-
 	return nil
 }
 
 func (c *OS2) Count(ctx context.Context, index, query string) (int64, error) {
 
-	resp, err := c.client.Count(c.client.Count.WithIndex(index), c.client.Count.WithBody(strings.NewReader(query)))
-
+	req := &osapi.IndicesCountReq{
+		Indices: []string{index},
+		Body:    strings.NewReader(query),
+	}
+	resp, err := c.client.Indices.Count(ctx, req)
 	if err != nil {
 		return 0, fmt.Errorf("OpenSearch Count: %w", err)
 	}
 
-	defer closeBody(resp)
-	if resp.IsError() {
-		return 0, c.parseError(resp)
-	}
-
-	type CountResponse struct {
-		Count int64 `json:"count"`
-	}
-
-	count := &CountResponse{}
-	if err := c.decoder.Decode(resp.Body, count); err != nil {
-		return 0, fmt.Errorf("decoding Opensearch Count result to int64: %w", err)
-	}
-
-	return count.Count, nil
+	return int64(resp.Count), nil
 }
 
 func (c *OS2) ClearScroll(ctx context.Context, scrollID string) error {
-	resp, err := c.client.ClearScroll(
-		c.client.ClearScroll.WithContext(ctx),
-		c.client.ClearScroll.WithScrollID(scrollID))
-
+	_, err := c.client.Scroll.Delete(ctx, osapi.ScrollDeleteReq{
+		ScrollIDs: []string{scrollID},
+	})
 	if err != nil {
 		return fmt.Errorf("OpenSearch ClearScroll: %w", err)
 	}
-
-	defer closeBody(resp)
-	if resp.IsError() {
-		return c.parseError(resp)
-	}
-
 	return nil
 }
 
 func (c *OS2) Scroll(ctx context.Context, index, body, scrollID string) (*client.Response, error) {
 
-	var resp *osapi.Response
-	var err error
+	var scrollResp *osapi.ScrollGetResp
+	var searchResp *osapi.SearchResp
+	var osResponse response
+	var searchErr error
+	// handle scroll id get call
 	if len(scrollID) != 0 {
-		resp, err = c.client.Scroll(
-			c.client.Scroll.WithScrollID(scrollID),
-			c.client.Scroll.WithScroll(time.Minute),
-			c.client.Scroll.WithContext(ctx),
-		)
+		scrollResp, searchErr = c.client.Scroll.Get(ctx, osapi.ScrollGetReq{
+			ScrollID: scrollID,
+			Params: osapi.ScrollGetParams{
+				Scroll: time.Minute,
+				// do not set scroll ID here as it will be added to the params and scroll ID can be excessively long
+			},
+		})
+		bodyBytes, err := io.ReadAll(scrollResp.Inspect().Response.Body)
```

Comment:
unfortunately looks like this needs to check `searchErr` here:
```
// do calls the opensearch.Client.Do() and checks the response for openseach api errors
func (c *Client) do(ctx context.Context, req opensearch.Request, dataPointer any) (*opensearch.Response, error) {
	resp, err := c.Client.Do(ctx, req, dataPointer)
	if err != nil {
		return nil, err
	}
```
AFAICT ^ that means that a request error will lead to a nil `Inspect().Response`, so `.Body` can panic